pkg/controller/account/account_controller.go

```diff
@@ -605,6 +604,27 @@ func CreateAccount(reqLogger logr.Logger, client awsclient.Client, accountName,
 	return accountStatus, nil
 }
 
+func checkAWSAccountsLimitReached(r *ReconcileAccount, reqLogger logr.Logger, currentAccounts int) (bool, error) {
+	instance := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: awsv1alpha1.AccountCrNamespace, Name: awsv1alpha1.DefaultConfigMap}, instance)
+	if err != nil {
+		unexpectedErrorMsg := fmt.Sprintf("%s: Failed to retrieve default ConfigMap, account limit defaulting to 100", awsv1alpha1.ErrMissingDefaultConfigMap)
+		reqLogger.Info(unexpectedErrorMsg)
+	} else {
+		if limit, ok := instance.Data["account-limit"]; ok {
+			if i, err := strconv.Atoi(limit); err == nil {
+				return i <= currentAccounts, nil
+			}
+			unexpectedErrorMsg := fmt.Sprintf("Account: Failed to convert ConfigMap 'account-limit' string field to int, account limit defaulting to 100")
```

Comment:
You don't need this or the else below. If the error != nil, then the code will continue, so you can put this line and the reqlogger at the end of the else block from 613

Reply:
@fahlmant Actually this logs when converting the `account-limit` field fails while the else bellow logs when the `account-limit` field is missing entirely. Since they are logging two different issues I think leaving it this way is best. Are you concerned about too many logs?

Reply:
Ah I see the different errors. I don't see a big difference between both errors, as they both say that the config map is misconfigured essentially. I guess it doesn't matter too much. Curious on other's thoughts

Reply:
It probably wouldn't hurt to be more explicit with the errors, to help anyone who may be on-call who's debugging a potential problem in the future who's not as familiar with the operator.

I do, however, like the idea of returning more often rather than these else blocks, as I've indicated in another comment earlier.

Reply:
@iamkirkbater Currently the method returns twice; when it works and we know exactly what the account limit is or when it fails and we want to default that limit to 100. Normally we would return at every error but since any other return would be nearly identical to the one that is already there I decided to return only once. We don't need to worry about "missing" an error since this method runs in a linearly fashion the `error` variable always has the newest value. This is why I approached this the way I did, logging any issues that were encounter and return only once.